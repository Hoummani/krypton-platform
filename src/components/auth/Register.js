/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/no-unescaped-entities */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Navbar from '../Core/Navbar';

class Register extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="register">
        <Navbar />
        <br />
        <div className="container">
          <div className="card">
            <div className="card-content">
              <div className="content">
                <h1 className="title">Join to krypton community</h1>
                <p className="">
                  if you already have one it's simple to{' '}
                  <a href="#">sign in now</a>
                </p>
                <div className="container">
                  <div className="field is-horizontal">
                    <div className="field-body">
                      <div className="field">
                        <p className="control is-expanded has-icons-left">
                          <input
                            className="input"
                            type="text"
                            id="firstName"
                            placeholder="FirstName"
                          />
                          <span className="icon is-small is-left">
                            <i className="fas fa-user" />
                          </span>
                        </p>
                      </div>
                      <div className="field">
                        <p className="control is-expanded has-icons-left">
                          <input
                            className="input"
                            type="text"
                            id="lastName"
                            placeholder="LastName"
                          />
                          <span className="icon is-small is-left">
                            <i className="fas fa-user" />
                          </span>
                        </p>
                      </div>
                    </div>
                  </div>

                  <div className="field -s-horizontal">
                    <div className="field body">
                      <div className="field">
                        <p className="control is-expanded has-icons-left has-icons-right">
                          <input
                            className="input"
                            type="email"
                            placeholder="Email"
                            defaultValue="alex@smith.com"
                          />
                          <span className="icon is-small is-left">
                            <i className="fas fa-envelope" />
                          </span>
                        </p>
                      </div>
                    </div>
                  </div>

                  <div className="field is-horizontal">
                    <div className="field-body">
                      <div className="field">
                        <p className="control is-expanded has-icons-left">
                          <input
                            className="input"
                            type="password"
                            id="password"
                            placeholder="Password"
                          />
                          <span className="icon is-small is-left">
                            <i className="fas fa-lock" />
                          </span>
                        </p>
                      </div>
                      <div className="field">
                        <p className="control is-expanded has-icons-left">
                          <input
                            className="input"
                            type="password"
                            id="confirmPassword"
                            placeholder="ConfirmPassword"
                          />
                          <span className="icon is-small is-left">
                            <i className="fas fa-lock" />
                          </span>
                        </p>
                      </div>
                    </div>
                  </div>

                  <div className="field is-horizontal">
                    <div className="field-body">
                      <div className="field">
                        <div className="control">
                          <button
                            className="button is-primary"
                            style={{ marginRight: '3px' }}
                            type="button"
                          >
                            <span className="icon">
                              <i className="fas fa-sign-in-alt" />
                            </span>
                            <span>Register</span>
                          </button>
                          <button className="button is-light" type="button">
                            <span className="icon">
                              <i className="fas fa-arrow-left" />
                            </span>
                            <span>
                              <Link className="has-text-black" to="/">
                                Go Back
                              </Link>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                  {/** 
                  <div className="notification is-warning">
                    <button className="delete" />
                    Don't tick the box of remember me if you are using a public or
                    shared computer
                  </div>
                  */}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Register;
