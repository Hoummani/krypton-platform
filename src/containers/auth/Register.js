/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/no-unescaped-entities */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-useless-constructor */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Navbar from '../../components/Core/Navbar';

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      confirmPassword: ''
    };
  }

  handleChange = e => {
    this.setState({
      [e.target.id]: e.target.value
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    // eslint-disable-next-line no-console
    console.log(this.state);
  };

  render() {
    return (
      <div className="register">
        <Navbar />
        <br />
        <div className="container">
          <div className="card">
            <form onSubmit={this.handleSubmit}>
              <div className="card-content">
                <div className="content">
                  <h1 className="title">Join to krypton platform</h1>
                  <p className="">
                    if you already have one it's simple to{' '}
                    <a href="#">sign in now</a>
                  </p>
                  <div className="container">
                    <div className="field is-horizontal">
                      <div className="field-body">
                        <div className="field">
                          <p className="control is-expanded has-icons-left">
                            <input
                              className="input"
                              type="text"
                              id="firstName"
                              onChange={this.handleChange}
                              placeholder="FirstName"
                              required
                            />
                            <span className="icon is-small is-left">
                              <i className="fas fa-user" />
                            </span>
                          </p>
                        </div>
                        <div className="field">
                          <p className="control is-expanded has-icons-left">
                            <input
                              className="input"
                              type="text"
                              id="lastName"
                              onChange={this.handleChange}
                              placeholder="LastName"
                              required
                            />
                            <span className="icon is-small is-left">
                              <i className="fas fa-user" />
                            </span>
                          </p>
                        </div>
                      </div>
                    </div>

                    <div className="field -s-horizontal">
                      <div className="field body">
                        <div className="field">
                          <p className="control is-expanded has-icons-left has-icons-right">
                            <input
                              className="input"
                              type="email"
                              onChange={this.handleChange}
                              placeholder="Email"
                              required
                            />
                            <span className="icon is-small is-left">
                              <i className="fas fa-envelope" />
                            </span>
                          </p>
                        </div>
                        <div className="field body">
                          <div className="field">
                            <p className="control is-expanded has-icons-left has-icons-right">
                              <input
                                className="input"
                                type="phone"
                                onChange={this.handleChange}
                                placeholder="Phone"
                                required
                              />
                              <span className="icon is-small is-left">
                                <i className="fas fa-phone" />
                              </span>
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>

                    <div className="field is-horizontal">
                      <div className="field-body">
                        <div className="field">
                          <p className="control is-expanded has-icons-left">
                            <input
                              className="input"
                              type="password"
                              onChange={this.handleChange}
                              id="password"
                              placeholder="Password"
                              required
                            />
                            <span className="icon is-small is-left">
                              <i className="fas fa-lock" />
                            </span>
                          </p>
                        </div>
                        <div className="field">
                          <p className="control is-expanded has-icons-left">
                            <input
                              className="input"
                              type="password"
                              id="confirmPassword"
                              onChange={this.handleChange}
                              placeholder="ConfirmPassword"
                              required
                            />
                            <span className="icon is-small is-left">
                              <i className="fas fa-lock" />
                            </span>
                          </p>
                        </div>
                      </div>
                    </div>

                    <div className="field is-horizontal">
                      <div className="field-body">
                        <div className="field">
                          <div className="control">
                            <button
                              className="button is-primary"
                              style={{ marginRight: '3px' }}
                              onChange={this.handleChange}
                              type="submit"
                              required
                            >
                              <span className="icon">
                                <i className="fas fa-sign-in-alt" />
                              </span>
                              <span>
                                <strong>Register</strong>
                              </span>
                            </button>
                            <button className="button is-light" type="button">
                              <span className="icon">
                                <i className="fas fa-arrow-left" />
                              </span>
                              <span>
                                <Link className="has-text-black" to="/">
                                  <strong>Go Back</strong>
                                </Link>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    {/** 
                    <div className="notification is-warning">
                      <button className="delete" />
                      Don't tick the box of remember me if you are using a public or
                      shared computer
                    </div>
                    */}
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default Register;
