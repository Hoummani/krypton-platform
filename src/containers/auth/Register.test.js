/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Register from './Register';

describe('Testing of Register component', () => {
  let component;
  let props;

  Enzyme.configure({
    adapter: new Adapter()
  });
  it('Should rendred the component !', () => {
    props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn()
    };
    component = shallow(
      <Register
        handleChange={props.handleChange}
        handleSubmit={props.handleSubmit}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
