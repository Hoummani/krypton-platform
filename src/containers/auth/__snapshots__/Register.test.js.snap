// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of Register component Should rendred the component ! 1`] = `ShallowWrapper {}`;
